code/__DATASTRUCTURES/linked_lists.dm

```diff
@@ -0,0 +1,173 @@
+
+//Ok so it's technically a double linked list, bite me.
+
+/datum/linked_list
+	var/list/contents = list()
```

Comment:
This isn't a real linked list, it is more like a [ring buffer](http://gameprogrammingpatterns.com/event-queue.html#a-ring-buffer) with the front and back pointers.

A real linked list would not have an underlying array (list), it would have nodes which point to nodes; while keeping track of the front and (maybe) back node.